motionplan/plan.go

```diff
@@ -0,0 +1,154 @@
+package motionplan
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	pb "go.viam.com/api/service/motion/v1"
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type Plan struct {
+	Trajectory
+	Path
```

Comment:
Rather than containing a `Trajectory` and a `Path`, which as a public struct should be made to be unrelated to one another, it might make more sense to have this contain a `Trajectory` and allow it to be queried to produce a `Path`, i.e. enshrine the `Trajectory` as the single source of truth.

Reply:
```
type Trajectory struct {
   map[string][]referenceframe.Input
   path
}
```
I think it we use a struct we should use a constructor.
I am worried that there will be an instance where `path` field of the `Trajectory` struct is nil.
thoughts?

Reply:
Yes we would need to create this using constructors since the members would not be public (this was a hasty mock up)